src/App.tsx

```diff
@@ -63,53 +72,104 @@ const Themes = {
   },
 } as const;
 
+export interface IBookmarkContext {
+  bookmarks: Map<string, IBookmark>;
+  onBookmark: (run: IRun) => void;
+}
+
+export const BookmarkContext = createContext<IBookmarkContext | null>(null);
+
 function App() {
-  const {error, data: events} = useSWR(
-    'https://api.submissions.esamarathon.com/events',
-    LoadEvents,
-  );
-  const [currentEventID, setCurrentEventID] = usePersistent<string | undefined>('preferred_event');
+  const {error, data: events, isValidating} = useSWR('/events', loadFromESASubmissions);
+  const [selectedEventID, setSelectedEvent] = usePersistent<string | undefined>('preferred_event');
+  const [bookmarkContext, setBookmarkContext] = useState<IBookmarkContext>(() => ({
+    bookmarks: GetBookmarks(),
+    onBookmark,
+  }));
 
-  SplashScreen.hide();
+  async function onBookmark(run: IRun) {
+    if (!run.id) {
+      console.warn('Trying to bookmark a run without an ID');
+      return;
+    }
 
-  if (!events) {
+    const isBookmarked = IsBookmarked(run.id);
+    if (isBookmarked) {
+      RemoveBookmark(run.id);
+
+      const bookmark = GetBookmark(run.id);
+      if (bookmark) {
+        await cancelNotification(bookmark.notificationId);
+      }
+    } else {
+      const {notifications} = await scheduleNotification({
+        title: 'Your run is about to start!',
+        body: `In about an hour ${run.game} - ${run.category} starts`,
+        scheduled: dayjs(run.scheduled).subtract(1, 'hour').toDate(),
+      });
+
+      StoreBookmark({run, notificationId: notifications[0].id});
```

Comment:
Curious, why are these functions named with capital letters?

Reply:
Idk, was like that before. I too prefer lowercase for non components.